src/point.jl

```diff
@@ -2,13 +2,49 @@
 "Abstract type for ASPRS LAS point data record formats 0 - 3"
 abstract type LasPoint end
 
-function Base.show(io::IO, pointdata::Vector{T}) where T <: LasPoint
+"""
+Custom PointVector struct for memory mapped LasPoints.
+Inspiration taken from UnalignedVector.jl
+and extended it with custom indexing and packing.
+"""
+struct PointVector{T} <: AbstractArray{T,1}
+    data::Vector{UInt8}
+    n::Int
+    pointsize::Int
+
+    function PointVector{T}(data::Vector{UInt8}) where {T}
+        pointsize = packed_sizeof(T)
+        n = length(data) ÷ pointsize
+        n*pointsize == length(data) || throw(DimensionMismatch("length(data) should be a multiple of $(pointsize), got $(length(data))"))
+        new{T}(data, n, pointsize)
+    end
+end
+
+Base.IndexStyle(::Type{PointVector{T}}) where {T} = IndexLinear()
+Base.length(pv::PointVector) = pv.n
+Base.size(pv::PointVector) = (length(pv),)
+
+function Base.getindex(pv::PointVector{T}, i::Int) where {T}
+    offset = (i-1) * pv.pointsize + 1
+    r = offset:offset + pv.pointsize - 1
+    io = IOBuffer(pv.data[r])
```

Comment:
Presumably this allocates? If you can figure out how to get an IO which points directly to the underlying array the performance should be a fair bit better.  Same comment for setindex.

Reply:
I assume so, I'll look into it. Probably looking at some sort of `reinterpret` then.

Reply:
Possibly.  In julia-0.7 there's a `GenericIOBuffer` which I suspect gives you this capability, but I think it's not available in 0.6.

Reply:
Ah, I think the same thing is available in 0.6 under the name `AbstractIOBuffer`.  In principle you'd just wrap the whole of `data` in an `IOBuffer`, `seek` to the desired `offset` and read the point rather than slicing the array.  That should involve zero allocation. I had a quick peek at the `code_native` for `seek(::IOBuffer, offset)` which a lot worse than I'd have hoped for. Still worth trying though.